packages/www/pages/verify.jsx

```diff
@@ -14,9 +14,28 @@ import Guides from "components/Marketing/Guides";
 const Verify = () => {
   useLoggedIn();
   const router = useRouter();
-  const { user } = useApi();
+  const { user, verify } = useApi();
   const { email, emailValidToken } = router.query;
 
+  useEffect(() => {
+    if (email && emailValidToken) {
+      verify(email, emailValidToken).then(() => {
+        if (selectedPlan === "1") {
+          router.replace("/dashboard/billing/plans?promptUpgrade=true");
+        } else {
+          router.replace("/dashboard");
+        }
+      });
+    }
+  }, [email, emailValidToken]);
+
+  // If they've already validated their email, get 'em out of here
+  useEffect(() => {
+    if (user && user.emailValid !== false) {
```

Comment:
**nit:**
```suggestion
    if (user?.emailValid === true) {
```
since `user.emailValid: undefined` would also pass the current test and we probably don't want undefined/null (or anything not a boolean) to be considered a validated user.